### ISSUE TITLE
String Literal in Method

### ISSUE THREAD

**Issue by sunnyrjuneja** [castwide/solargraph#24]
Hey @castwide,

Incredibly impressive work. Saw this gem mentioned in VS Code Ruby. Installed and it's working like a charm! I wanted to do some tests to see what is and what isn't supported. Something surprising I found was that String literals in method definitions don't seem to be producing results.

```
def hello
  "hi".
end
```

```
$ solargraph suggest play.rb --line 2 --column 7
WARNING: The `solargraph suggest` command is a candidate for deprecation. It will either change drastically or not exist in a future version.
(string):3:4: error: unexpected token $end
(string):3: end
(string):3:    
Invalid offset
/home/wasd/.gem/ruby/2.4.0/gems/solargraph-0.16.0/lib/solargraph/api_map/source.rb:384:in `get_position_at'
/home/wasd/.gem/ruby/2.4.0/gems/solargraph-0.16.0/lib/solargraph/code_map.rb:105:in `comment_at?'
/home/wasd/.gem/ruby/2.4.0/gems/solargraph-0.16.0/lib/solargraph/code_map.rb:199:in `suggest_at'
/home/wasd/.gem/ruby/2.4.0/gems/solargraph-0.16.0/lib/solargraph/shell.rb:59:in `suggest'
/home/wasd/.gem/ruby/2.4.0/gems/thor-0.20.0/lib/thor/command.rb:27:in `run'
/home/wasd/.gem/ruby/2.4.0/gems/thor-0.20.0/lib/thor/invocation.rb:126:in `invoke_command'
/home/wasd/.gem/ruby/2.4.0/gems/thor-0.20.0/lib/thor.rb:387:in `dispatch'
/home/wasd/.gem/ruby/2.4.0/gems/thor-0.20.0/lib/thor/base.rb:466:in `start'
/home/wasd/.gem/ruby/2.4.0/gems/solargraph-0.16.0/bin/solargraph:5:in `<top (required)>'
/home/wasd/.gem/ruby/2.4.0/bin/solargraph:23:in `load'
/home/wasd/.gem/ruby/2.4.0/bin/solargraph:23:in `<main>'
{"status":"err","message":"Invalid offset\n/home/wasd/.gem/ruby/2.4.0/gems/solargraph-0.16.0/lib/solargraph/api_map/source.rb:384:in `get_position_at'\n/home/wasd/.gem/ruby/2.4.0/gems/solargraph-0.16.0/lib/solargraph/code_map.rb:105:in `comment_at?'\n/home/wasd/.gem/ruby/2.4.0/gems/solargraph-0.16.0/lib/solargraph/code_map.rb:199:in `suggest_at'\n/home/wasd/.gem/ruby/2.4.0/gems/solargraph-0.16.0/lib/solargraph/shell.rb:59:in `suggest'\n/home/wasd/.gem/ruby/2.4.0/gems/thor-0.20.0/lib/thor/command.rb:27:in `run'\n/home/wasd/.gem/ruby/2.4.0/gems/thor-0.20.0/lib/thor/invocation.rb:126:in `invoke_command'\n/home/wasd/.gem/ruby/2.4.0/gems/thor-0.20.0/lib/thor.rb:387:in `dispatch'\n/home/wasd/.gem/ruby/2.4.0/gems/thor-0.20.0/lib/thor/base.rb:466:in `start'\n/home/wasd/.gem/ruby/2.4.0/gems/solargraph-0.16.0/bin/solargraph:5:in `<top (required)>'\n/home/wasd/.gem/ruby/2.4.0/bin/solargraph:23:in `load'\n/home/wasd/.gem/ruby/2.4.0/bin/solargraph:23:in `<main>'"}

```

Inside of VS code, I am seeing fairly generic suggestions. It looks like stuff defined in Kernel and keywords.

**Comment by castwide** [comment]
Thanks, @sunnyrjuneja. One thing to note: the `suggest` command expects the line and the column to be zero-based, so you probably want `--line` to be 1 in that test.

Nevertheless, you've definitely found a bug. It's been addressed on the master branch and will work correctly in the next version.

**Comment by sunnyrjuneja** [comment]
That makes sense! Thank you so much.

One unrelated note, I would love to help contribute (I've read the most recent commits and browsed issues / PRs) but I'm not sure what you're planning / in scope / out of scope / needs to be completed. Let me know if you decide to write these down or want to chat about it.

**Comment by castwide** [comment]
Thanks for your interest. I have a project for Solargraph in Pivotal Tracker: https://www.pivotaltracker.com/n/projects/1970245

I just finished wrapping up the last feature for 0.17.0, which I plan to release tomorrow. If you have any recommendations or issues you'd like to tackle, I'm open to discussion.

**Comment by castwide** [comment]
This issue is fixed in version 0.17.0.

**Comment by sunnyrjuneja** [comment]
Excellent, I'll close the issue.
